### ISSUE TITLE
Too Broad Github Permissions are being asked

### ISSUE THREAD

**Issue by UdayKiranPadhy** [raphaelheinz/LeetHub-3.0#33]
## Please review previous closed issues before filling out a new one! Duplicate issues will be closed without comment.

**Is your feature request related to a problem? Please describe.**
When we authorize this application to github , it is asking read / write for all public and private repos. Similar to Admin Access
It would be much better to reduce to scope of the application to read and write to selected repos or use fine grained Access tokens.

**Comment by raphaelheinz** [comment]
Hi @UdayKiranPadhy,
thanks, I agree this is a good idea. I am not yet sure how to reduce the scope to specific repos, because the plugin would need access to list them in the first place. Will need to read some docs to get more information :) Have you already worked with Github autorization?

**Comment by UdayKiranPadhy** [comment]
Yeah i have worked on Github Authorization previously , and i think its definitely possible to achieve this.
